docs/adr-0001-publishing_metric_docs.md

```diff
@@ -0,0 +1,84 @@
+# Publishing documentation for metrics
+
+* Status: proposed <!-- [proposed | rejected | accepted | deprecated | … | superseded by [ADR-0005](0005-example.md)] optional -->
+* Deciders: emtwo, scholtzan, tdsmith, TBD
+* Date: 2020-11-19
+
+Technical Story: https://github.com/mozilla/jetstream/issues/296
+
+## Context and Problem Statement
+
+We need to present some context for our metrics on dashboards and use friendlier names than the column labels we have in our output today.
+We've already decided that this context should live next to the metric definitions, either in mozanalysis or in jetstream-config.
+We haven't decided how to make them accessible to downstream consumers like Experimenter and Partybal yet.
+
+Jetstream is the point in the system that understands which metrics will be included in an analysis
+and has access to the context defined with the metrics,
+so it makes sense for Jetstream to be responsible for collating those definitions.
+
+## Decision Drivers
+
+* We should minimize the number of systems that downstream consumers have to contend with.
+* ?
+
+## Considered Options
+
+* dictionary.json
+* Google Data Catalog
+* Something else
+
+## Decision Outcome
+
+TK
+
+<!-- Chosen option: "[option 1]", because [justification. e.g., only option, which meets k.o. criterion decision driver | which resolves force force | … | comes out best (see below)].
+
+### Positive Consequences
+
+* [e.g., improvement of quality attribute satisfaction, follow-up decisions required, …]
+* …
+
+### Negative Consequences
+
+* [e.g., compromising quality attribute, follow-up decisions required, …]
+* …
+
+-->
+
+## Pros and Cons of the Options <!-- optional -->
+
+### "dictionary.json": Save some JSON to GCS
```

Comment:
We could call this `metadata_<experiment_slug>.json`, as a place for metadata associated with the experiment. Some of this metadata are the metrics descriptions, but I think there is a possibility that additional metadata might be added in the future. So we'd have something like:

```json
{
  "metrics": {
      "active_hours": {
        "friendly_name": "Active hours",
        "description": "Measures the amount of time (in 5-second increments) during which Firefox received user input from a keyboard or mouse. The Firefox window does not need to be focused.",
        "bigger_is_better": true
    },...
  },
  "last_updated": "2020-11-19T12:32:11"  // some other future metadata
}
```

Reply:
@jaredlockhart also mentioned exposing the relationship between a probe set and its summaries

Reply:
`metadata_<experiment_slug>.json` sounds good to me. Though, I am wondering because of the existing data at the moment if it would be redundant, for example, to define `active_hours` in a separate file for every single experiment even though the definition is the same. Wondering if you've considered having 1 file with all the metrics in it. Perhaps it doesn't work for some reason but it would be worth noting it in the ADR I think.

Reply:
That sounds like a good contrast.

I can think of a couple of reasons I don't prefer it:
- Metrics can be changed or removed from mozanalysis -- I see pros and cons to whether the dashboard should capture the definition at the time of analysis or a definition that was updated since the experiment was analyzed; the latter could sometimes be incorrect, which I think is worse than making it hard to fix typos or stale information.
- Regenerating the list of all known metrics requires comprehending all experiment configurations that have ever been processed; without some extra work, it also implies a new requirement that experiment configurations should never use a metric name that has been used before with a different definition, which I think is too constraining.

I think the chief advantage of one-file-to-rule-them-all (or a database) is that it avoids some redundancy, but I don't think the storage costs are important.

Do you see any pros or cons that I'm missing?

Reply:
Nice. Yea the main reason I was suggesting the one file idea was for avoiding redundancy and so you wouldn't need to go back and make changes to all the files if a definition changed.

But it sounds like my conception of a metric's lifecycle wasn't accurate. Sounds like a metric can be changed/removed and in those cases it's better to not retroactively update old definitions. In which case, I can see why the mult-file approach is better.

Reply:
Added as a contrast.